### DISCUSSION TITLE
Generate text on nth line

### DISCUSSION THREAD

**Discussion by s1n7ax** [L3MON4D3/LuaSnip#1184] [Q&A] [answered]
I have following component to generate the `React` import statement if it does not exists. However, I would like to have the import statement at the top of the file (always in the first line). Just wondering if this is possible in luasnip?

https://github.com/s1n7ax/nvim-snips/blob/f1903ae173d98f6a4c82d0b44e644c0eaa0c6340/lua/snips/javascriptreact/primitives/component.lua

https://github.com/L3MON4D3/LuaSnip/assets/18459807/e78027c8-8e5c-440e-9c98-210fd65c58ca

**Comment by L3MON4D3** [comment, marked as answer]
I'll have to re-check the code, but you may get away with calling `nvim_buf_set_text` in the functionNode to insert text anywhere in the buffer.. could you try if it works correctly?

**Comment by s1n7ax** [comment]
Yep. Works beautifully.

```lua
return function()
	return fmt(
		[[
		{}export default function {}() {{
			return({})
		}}
		]],
		{
			f(function()
				if not ts_react:has_react_import() then
					vim.api.nvim_buf_set_lines(0, 0, 0, true, {
						'import React from "react"',
					})
				end

				return {}
			end, {}),
			i(1, 'Name'),
			i(2, ''),
		}
	)
end

```
